examples/resolver/resolver.go

```diff
@@ -0,0 +1,28 @@
+package resolver
+
+import (
+	"fmt"
+	"net/http"
+)
+
+//RequestURL simply returns
+//the request url from REQUEST_URI header
+//this should not be done in production applications
+type RequestURL struct {
```

Comment:
I could do a CallbackResolver so that people would only need to implement a callback function instead of a struct with two methods, but that's about it :/